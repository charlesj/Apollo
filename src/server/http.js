
export const HttpConstants = {
  BadRequestName: 'BadRequest',
  NotFoundName: 'NotFound',
  NotFoundMessage: 'Resource not found',
  ServerErrorMessage: 'The server encountered an error processing the request',
  ServerErrorName: 'InternalServerError',
  UnauthorizedMessage: 'The user is not authorized to do this',
  UnauthorizedName: 'Unauthorized',
}

export function accepted(res, message) {
  res.status(202).json({ message })
}

export function badRequest(res, errorMessage, validationErrors) {
  res.status(400).json({
    name: HttpConstants.BadRequestName,
    errorMessage,
    validationErrors,
  })
}

export function created(res, entity) {
  res.status(201).json(entity)
}

export function noContent(res) {
  res.status(204).end()
}

export function notFound(res) {
  res.status(404).json({
    message: HttpConstants.NotFoundMessage,
    name: HttpConstants.NotFoundName,
  })
}

export function ok(res, entity) {
  res.status(200).json(entity)
}

export function serverError(res) {
  res.status(500).json({
    message: HttpConstants.ServerErrorMessage,
    name: HttpConstants.ServerErrorName,
  })
}


export function unauthorized(res) {
  res.status(401).json({
    message: HttpConstants.UnauthorizedMessage,
    name: HttpConstants.UnauthorizedName,
  })
}
