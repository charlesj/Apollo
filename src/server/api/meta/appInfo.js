import { ok } from '../../http'
import { handleError } from '../../handleError'

export const appInfo = handleError((req, res) => {
  ok(res, {
    version: '1.0.0',
  })
})



