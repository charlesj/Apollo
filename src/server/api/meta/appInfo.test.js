import { appInfo } from './appInfo'
import { mockResponse, mockRequest } from '../../../tests/expressMocks'
import { ok } from '../../http'

jest.mock('../../http')

let req = mockRequest()
let res = mockResponse()
let next = jest.fn()

describe('appInfo', () => {
  it('returns expected values', () => {
    appInfo(req, res, next)
    expect(ok).toHaveBeenCalledTimes(1)
    expect(ok.mock.calls[0][0]).toBe(res)
    expect(ok.mock.calls[0][1]).toMatchSnapshot()
  })
})
