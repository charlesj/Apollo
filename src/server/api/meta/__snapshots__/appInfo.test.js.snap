// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appInfo returns expected values 1`] = `
Object {
  "version": "1.0.0",
}
`;
